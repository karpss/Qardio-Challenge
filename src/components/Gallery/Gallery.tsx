/* eslint-disable */
import React, {useEffect, useReducer} from 'react';
import axios from 'axios';
import reducer from '../../reducer/Reducer';
import './Gallery.css';

const Gallery = () => {
    const [{data, isLoading, error}, dispatch] = useReducer(reducer, {
        isLoading: false,
      });
    
      useEffect(() => {
        let ignore = false;
    
        dispatch({type: 'request'});
        axios(
          `https://www.rijksmuseum.nl/api/nl/collection?key=yW6uq3BV&involvedMaker=Rembrandt+van+Rijn`
        ).then(
          (results) => {
            if (!ignore) dispatch({type: 'success', results: results.data});
          },
          (error: string) => dispatch({type: 'failure', error})
        );
    
        return (): void => {
          ignore = true;
        };
      }, []);
    
      // console.log(data);
    
      return (
        <>
          <h1>MUSEUM</h1>
          <div className="container">
            {isLoading && <span>Loading...</span>}
            {error && <span>Error: {error}</span>}
            <ul>
              {data &&
                data.artObjects &&
                data.artObjects.map((item) => (
                  <div className="card" key={item.id}>
                    <img
                      src={item.webImage.url}
                      alt="Gallery Images"
                    />
                    <p>{item.longTitle}</p>
                  </div>
                ))}
            </ul>
          </div>
        </>
      );
}

export default Gallery;