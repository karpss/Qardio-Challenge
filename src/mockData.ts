/* eslint-disable */
export const mockData = {
    
    artObjects: {
        id: 'nl-SK-A-4050',
        longTitle: "Zelfportret als de apostel Paulus, Rembrandt van Rijn, 1661",
        webImage: {
            url:"https://lh3.googleusercontent.com/NrCcfeY0r2F3M2hIQe5SLDRofR2tVzeOH18VjflOYGj88v4clb4v2H_VgCZR4nJhYsxxH9ATzfkL2tRqOWEK5-gPVEE=s0"
        }
    
    }
      

}

