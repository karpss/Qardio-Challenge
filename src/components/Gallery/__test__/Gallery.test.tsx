/* eslint-disable */
import React, { render, screen } from "@testing-library/react";
import { mockData } from "../../../mockData";
import Gallery from "../Gallery";

test("renders Museum Header", () => {
  render(<Gallery />);
  const headerElement = screen.getByText(/MUSEUM/i);
  expect(headerElement).toBeInTheDocument();
});

test("shows Loading message while fetching request", () => {
  render(<Gallery />);
  const loadingMessage = screen.getByText(/Loading.../i);
  expect(loadingMessage).toBeInTheDocument();
});

test("matches snapshot", () => {
  const { container } = render(<Gallery {...mockData} />);
  expect(container.firstChild).toMatchSnapshot();
});
