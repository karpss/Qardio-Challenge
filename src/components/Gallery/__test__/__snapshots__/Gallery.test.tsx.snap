// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h1>
  MUSEUM
</h1>
`;
